---
int8 num_objects
int8[] ids
string[] names
